File: src/FizzBuzz.java
import java.util.ArrayList;
import java.util.List;
import java.util.function.Consumer;

/**
 * Name: FizzBuzz
 * Author: lloydfinch
 * Function:
 * 写一个程序，输出从 1 到 n 数字的字符串表示。
 * <p>
 * 1. 如果 n 是3的倍数，输出“Fizz”；
 * <p>
 * 2. 如果 n 是5的倍数，输出“Buzz”；
 * <p>
 * 3.如果 n 同时是3和5的倍数，输出 “FizzBuzz”。
 * <p>
 * 来源：力扣（LeetCode）
 * 链接：https://leetcode-cn.com/problems/fizz-buzz
 * 著作权归领扣网络所有。商业转载请联系官方授权，非商业转载请注明出处。
 * Date: 2020-04-23 16:22
 * Modify: lloydfinch 2020-04-23 16:22
 */
public class FizzBuzz {

    public static void main(String[] args) {
        FizzBuzz fizzBuzz = new FizzBuzz();

        int n = 15;
        List<String> result = fizzBuzz.fizzBuzz(n);
        result.forEach(new Consumer<String>() {
            @Override
            public void accept(String s) {
                System.out.println(s);
            }
        });

    }

    public List<String> fizzBuzz(int n) {
        List<String> list = new ArrayList<>();

        for (int i = 1; i <= n; i++) {
            if (i % 3 == 0 && i % 5 == 0) {
                list.add("FizzBuzz");
            } else if (i % 3 == 0) {
                list.add("Fizz");
            } else if (i % 5 == 0) {
                list.add("Buzz");
            } else {
                list.add(String.valueOf(i));
            }
        }

        return list;
    }
}

File: src/kt/FizzBuzz.kt
package kt

/**
 *
 * Name: FizzBUzz
 * Author: lloydfinch
 * Function: //TODO
 * Date: 2020-04-23 16:29
 * Modify: lloydfinch 2020-04-23 16:29
 */
fun main() {
    val fizzBuzz = fizzBuzz(15)
    fizzBuzz.forEach {
        println(it)
    }
}

fun fizzBuzz(n: Int): List<String> {
    val list = mutableListOf<String>()

    for (i in 1..n) {
        list.add(when {
            i % 3 == 0 && i % 5 == 0 -> "FizzBuzz"
            i % 3 == 0 -> "Fizz"
            i % 5 == 0 -> "Buzz"
            else -> i.toString()
        })
    }

    return list
}